filters/filters.go

```diff
@@ -0,0 +1,52 @@
+package filters
+
+import (
+	"bytes"
+	"regexp"
+)
+
+func NewPrefixFilter(pfx string) func([]byte) []byte {
+	if len(pfx) == 0 {
+		return func(bs []byte) []byte {
+			return bs
+		}
+	}
+
+	return func(bs []byte) []byte {
+		buf := bytes.Buffer{}
+
+		if len(pfx) > 0 {
```

Comment:
If we already checked first to make sure that the length of the prefix is 0 for early return, is it necessary to do the check here? Could it just be simplified to have just this block and remove the early exit scenario?